streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -1961,10 +1920,10 @@ public void postCommit(final boolean enforceCheckpoint) {
         };
 
         // `handleAssignment`
-        expectRestoreToBeCompleted(consumer);
+        expectAssignmentToBeCalled(consumer);
         when(activeTaskCreator.createTasks(any(), Mockito.eq(taskId00Assignment))).thenReturn(singletonList(task00));
-        expect(consumer.assignment()).andReturn(taskId00Partitions);
-        replay(consumer);
+        final Set<TopicPartition> partitions = union(HashSet::new, taskId00Partitions);
+        when(consumer.assignment()).thenReturn(partitions);
```

Comment:
Why do you union a single set? 
```suggestion
        when(consumer.assignment()).thenReturn(taskId00Partitions);
```
... and isn't this a duplicate of line 1923 (except the differing partitions).
```java
expectAssignmentToBeCalled(consumer);
```
Could it be that you can remove line 1923?

Reply:
Because I couldn't figure out a better way to create a set with only one element and Collections does not have a singletonSet 😞

Reply:
Oh, silly me, there is a mkSet method in Utils. Okay, I will remedy this.

Reply:
You are correct, it appears I can remove line 1923